document-store/src/main/java/org/hypertrace/core/documentstore/query/SortingSpec.java

```diff
@@ -15,6 +17,7 @@
  */
 @Value
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
+@EqualsAndHashCode(cacheStrategy = CacheStrategy.LAZY)
```

Comment:
query: what's the benefit of this since the fields are not final?

Reply:
All the fields and the class are implicitly final through the `@Value` annotation above.
Since this is basically an immutable class, we cache the hashCode for optimization.

Reply:
As discussed, I'm removing these.